src/main/java/org/takes/rq/RqRequestLine.java

```diff
@@ -0,0 +1,214 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.rq;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.EqualsAndHashCode;
+import org.takes.HttpException;
+import org.takes.Request;
+
+/**
+ * HTTP Request-Line parsing.
+ *
+ * <p>All implementations of this interface must be immutable and thread-safe.
+ *
+ * @author Vladimir Maksimenko (xupypr@xupypr.com)
+ * @version $Id$
+ * @since 0.29.1
+ */
+public interface RqRequestLine extends Request {
+
+    /**
+     * Get Request-Line header.
+     * @return HTTP Request-Line header
+     * @throws IOException If fails
+     */
+    String requestLineHeader() throws IOException;
```

Comment:
@xupyprmv why not just `header`? there are many headers? we really need to say "request line header" in order not to confuse it with some other header? see http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html keep in mind, the longer the name - the more messy is your code. if you really need long names - it's a sign that your code needs refactoring